ecs-init/docker/docker.go

```diff
@@ -152,12 +165,24 @@ func (c *Client) getContainerConfig() *godocker.Config {
 	}
 }
 
-func (c *Client) loadEnvVariables() []string {
+func (c *Client) loadEnvVariables() map[string]string {
+	envVariables := make(map[string]string)
+
 	file, err := c.fs.ReadFile(config.AgentConfigFile())
 	if err != nil {
-		return make([]string, 0)
+		return envVariables
 	}
-	return strings.Split(strings.TrimSpace(string(file)), "\n")
+
+	lines := strings.Split(strings.TrimSpace(string(file)), "\n")
+	for _, line := range lines {
+		parts := strings.SplitN(strings.TrimSpace(line), "=", 2)
+		if len(parts) != 2 {
+			continue
```

Comment:
If you think this is important, I can create a PR to the dev branch to fix this and then update this PR.  Otherwise, I'd rather leave this until later.